spark-commons/src/main/scala/za/co/absa/spark/commons/implicits/DataFrameImplicits.scala

```diff
@@ -83,6 +85,68 @@ object DataFrameImplicits {
       }
     }
 
+    /**
+     * Casts NullType (aka VOID) fields to their target types as defined in `targetSchema`.
+     *
+     * Matching of fields is "by name", so order of fields in schema(s) doesn't matter.
+     * Resulting DataFrame has the same fields order as original DataFrame.
+     *
+     * @param targetSchema definition of field types to which potential NullTypes will be casted to
+     * @return DataFrame with fields of NullType casted to their type in `targetSchema`
+     */
+    def enforceTypeOnNullTypeFields(targetSchema: StructType): DataFrame = {
+
+      def processArray(
+                        thisArrType: ArrayType, targetArrType: ArrayType, thisArrayColumn: Column
+                      ): Column =
+        (thisArrType.elementType, targetArrType.elementType) match {
+          case (_: NullType, _) =>
+            transform(
+              thisArrayColumn,
+              _ => lit(null).cast(targetArrType.elementType)
+            )
+          case (thisNestedArrType: ArrayType, targetNestedArrType: ArrayType) =>
+            transform(
+              thisArrayColumn,
+              processArray(thisNestedArrType, targetNestedArrType, _)
+            )
+          case (thisNestedStructType: StructType, targetNestedStructType: StructType) =>
+            transform(
+              thisArrayColumn,
+              element => struct(processStruct(thisNestedStructType, targetNestedStructType, Some(element)): _*)
+            )
+          case _ => thisArrayColumn
+        }
+
+      def processStruct(
+                         currentThisSchema: StructType, currentTargetSchema: StructType, parent: Option[Column]
+                       ): List[Column] = {
+        val currentTargetSchemaMap = currentTargetSchema.map(f => (f.name.toLowerCase, f)).toMap
+
+        currentThisSchema.foldRight(List.empty[Column])((field, acc) => {
+          val currentColumn: Column = parent
+            .map(_.getField(field.name))
+            .getOrElse(col(field.name))
+            .as(field.name)
```

Comment:
This can be omitted, if the `.as` is added to the `_` case bellow (as it's present in all other cases as well.

Reply:
Yup, changed.